### ISSUE TITLE
jspm install jade results in EISDIR error

### ISSUE THREAD

**Issue by Rush** [jspm/jspm-cli#1368]
```
$ jspm --version
0.16.18
Running against local jspm install.
$ node --version
v4.2.3
$ jspm install jade           
     Updating registry cache...
     Looking up github:johnsoftek/plugin-jade
     Looking up npm:jade
     Downloading npm:jade@1.11.0
     Looking up npm:character-parser
     Looking up npm:clean-css
     Looking up npm:commander
     Looking up npm:constantinople
     Looking up npm:jstransformer
     Looking up npm:mkdirp
     Looking up npm:transformers
     Looking up npm:uglify-js
     Looking up npm:void-elements
     Looking up npm:with
     Looking up npm:minimist
     Looking up npm:acorn
     Looking up npm:acorn-globals
     Looking up npm:is-promise
     Looking up npm:promise
     Downloading npm:clean-css@3.4.8
     Looking up npm:source-map
     Looking up github:jspm/nodelibs-fs
     Looking up github:jspm/nodelibs-path
     Looking up github:jspm/nodelibs-process
     Looking up github:jspm/nodelibs-http
     Looking up github:jspm/nodelibs-assert
ok   Installed npm:character-parser@1.2.1 (1.2.1)
ok   Installed npm:with@~4.0.0 (4.0.3)
ok   Installed npm:jstransformer@0.0.2 (0.0.2)
     Downloading npm:acorn-globals@1.0.9
     Looking up npm:graceful-readlink
ok   Installed npm:mkdirp@~0.5.0 (0.5.1)
ok   Installed npm:void-elements@~2.0.1 (2.0.1)
     Looking up npm:amdefine
     Looking up npm:css
     Looking up npm:uglify-to-browserify
     Looking up npm:yargs
     Looking up npm:optimist
     Looking up npm:asap
     Looking up npm:async
     Looking up npm:path-browserify
     Looking up npm:punycode
     Looking up npm:readable-stream
     Looking up npm:util
     Looking up npm:window-size
     Downloading npm:uglify-js@2.6.1
     Looking up github:jspm/nodelibs-child_process
     Looking up npm:https-browserify
     Looking up github:jspm/nodelibs-events
     Looking up npm:md5
     Looking up npm:url
     Downloading npm:source-map@0.5.3
     Downloading npm:is-promise@2.1.0
ok   Installed npm:minimist@0.0.8 (0.0.8)
     Looking up github:jspm/nodelibs-vm
ok   Installed npm:commander@~2.8.0 (2.8.1)
ok   Installed npm:commander@~2.6.0 (2.6.0)
     Looking up github:jspm/nodelibs-util
     Looking up npm:stream-browserify
     Looking up npm:css-parse
     Looking up npm:css-stringify
ok   Installed npm:promise@~2.0 (2.0.0)
ok   Installed npm:transformers@2.1.0 (2.1.0)
ok   Installed npm:constantinople@~3.0.1 (3.0.2)
     Looking up npm:Base64
     Looking up npm:inherits
     Looking up npm:wordwrap
ok   Installed npm:uglify-js@~2.2.5 (2.2.5)
     Looking up github:jspm/nodelibs-module
ok   Installed npm:graceful-readlink (1.0.1)
ok   Installed npm:is-promise@~1 (1.0.1)
     Looking up npm:vm-browserify
ok   Installed npm:amdefine (1.0.0)
     Looking up github:jspm/nodelibs-stream
ok   Installed npm:promise@^6.0.1 (6.1.0)
ok   Installed npm:asap@~1.0.0 (1.0.0)
ok   Installed npm:css@~1.0.8 (1.0.8)
ok   Installed npm:source-map@~0.4.0 (0.4.4)
     Looking up github:jspm/nodelibs-url
     Downloading npm:acorn@2.6.4
ok   Installed github:jspm/nodelibs-child_process@^0.1.0 (0.1.0)
ok   Installed npm:async@~0.2.6 (0.2.10)
     Looking up npm:events
     Downloading npm:yargs@3.10.0
     Looking up npm:camelcase
     Looking up npm:cliui
     Looking up npm:decamelize
ok   Installed npm:source-map@~0.1.7 (0.1.34)
ok   Installed github:jspm/nodelibs-http@^1.7.0 (1.7.1)
ok   Installed github:jspm/nodelibs-vm@^0.1.0 (0.1.0)
ok   Installed npm:window-size@0.1.0 (0.1.0)
ok   Installed npm:url@^0.10.1 (0.10.3)
     Looking up npm:querystring
ok   Installed npm:punycode@1.3.2 (1.3.2)
     Looking up npm:assert
     Downloading github:jspm/nodelibs-process@0.1.2
     Looking up npm:indexof
ok   Installed npm:css-stringify@1.0.5 (1.0.5)
ok   Installed npm:optimist@~0.3.5 (0.3.7)
     Downloading npm:cliui@2.1.0
     Looking up npm:center-align
     Looking up npm:right-align
     Downloading npm:decamelize@1.1.1
ok   Installed github:jspm/nodelibs-events@^0.1.1 (0.1.1)
     Looking up github:jspm/nodelibs-tty
ok   Installed npm:inherits@~2.0.1 (2.0.1)
ok   Installed npm:path-browserify@0.0.0 (0.0.0)
ok   Installed npm:stream-browserify@^1.0.0 (1.0.0)
ok   Installed npm:readable-stream@^1.0.27-1 (1.1.13)
     Looking up npm:core-util-is
     Looking up npm:isarray
     Looking up npm:string_decoder
     Looking up npm:process
ok   Installed npm:Base64@~0.2.0 (0.2.1)
ok   Installed github:jspm/nodelibs-module@^0.1.0 (0.1.0)
     Looking up github:jspm/nodelibs-buffer
ok   Installed npm:css-parse@1.0.4 (1.0.4)
ok   Installed github:jspm/nodelibs-url@^0.1.0 (0.1.0)
ok   Installed github:jspm/nodelibs-util@^0.1.0 (0.1.0)
ok   Installed npm:util@^0.10.3 (0.10.3)
     Downloading npm:core-util-is@1.0.2
     Downloading npm:right-align@0.1.3
     Looking up npm:align-text
ok   Installed npm:acorn@^1.0.1 (1.2.2)
     Downloading npm:process@0.11.2
ok   Installed npm:uglify-to-browserify@~1.0.0 (1.0.2)
     Downloading npm:center-align@0.1.2
     Looking up npm:lazy-cache
ok   Installed npm:acorn-globals@^1.0.3 (1.0.9)
ok   Installed github:jspm/nodelibs-path@^0.1.0 (0.1.0)
ok   Installed github:jspm/nodelibs-assert@^0.1.0 (0.1.0)
ok   Installed github:jspm/nodelibs-stream@^0.1.0 (0.1.0)
     Looking up npm:tty-browserify
ok   Installed npm:vm-browserify@0.0.4 (0.0.4)
ok   Installed npm:wordwrap@~0.0.2 (0.0.2)
     Looking up npm:is-buffer
ok   Installed npm:events@1.0.2 (1.0.2)
ok   Installed npm:assert@^1.3.0 (1.3.0)
ok   Installed npm:indexof@0.0.1 (0.0.1)
ok   Installed npm:querystring@0.2.0 (0.2.0)
     Downloading npm:align-text@0.1.3
     Looking up npm:kind-of
     Looking up npm:longest
     Looking up npm:repeat-string
ok   Installed github:jspm/nodelibs-tty@^0.1.0 (0.1.0)
ok   Installed npm:string_decoder@~0.10.0 (0.10.31)
ok   Installed npm:isarray@0.0.1 (0.0.1)
ok   Installed npm:is-promise@^2.0.0 (2.1.0)
     Downloading npm:longest@1.0.1
ok   Installed npm:camelcase@^1.0.2 (1.2.1)
ok   Installed npm:tty-browserify@0.0.0 (0.0.0)
ok   Installed github:jspm/nodelibs-process@^0.1.0 (0.1.2)
     Looking up npm:buffer
     Downloading npm:repeat-string@1.5.2
     Downloading npm:kind-of@2.0.1
ok   Installed npm:decamelize@^1.0.0 (1.1.1)
ok   Installed npm:is-buffer@^1.0.2 (1.0.2)
     Downloading npm:buffer@3.5.5
     Looking up npm:base64-js
     Looking up npm:ieee754
     Downloading npm:isarray@1.0.0
ok   Installed github:jspm/nodelibs-buffer@^0.1.0 (0.1.0)
ok   Installed npm:right-align@^0.1.1 (0.1.3)
     Downloading npm:lazy-cache@0.2.7
ok   Installed npm:center-align@^0.1.1 (0.1.2)
ok   Installed npm:process@^0.11.0 (0.11.2)
ok   Installed npm:align-text@^0.1.1 (0.1.3)
ok   Installed npm:core-util-is@~1.0.0 (1.0.2)
ok   Installed npm:cliui@^2.1.0 (2.1.0)
ok   Installed npm:yargs@~3.10.0 (3.10.0)
ok   Installed npm:longest@^1.0.1 (1.0.1)
ok   Installed npm:ieee754@^1.1.4 (1.1.6)
ok   Installed npm:base64-js@0.0.8 (0.0.8)
ok   Installed npm:repeat-string@^1.5.2 (1.5.2)
ok   Installed npm:kind-of@^2.0.0 (2.0.1)
ok   Installed npm:lazy-cache@^0.2.4 (0.2.7)
ok   Installed npm:uglify-js@^2.4.19 (2.6.1)
ok   Installed npm:isarray@^1.0.0 (1.0.0)
ok   Installed npm:source-map@~0.5.1 (0.5.3)

err  Error: EISDIR: illegal operation on a directory, read
         at Error (native)

warn Installation changes not saved.
```

**Comment by Rush** [comment]
I managed to nail it down to:
```
[pid 105022] unlink("<PROJECT_DIR>/jspm_packages/npm/css-stringify@1.0.5") = -1 EISDIR (Is a directory)
```

**Comment by Rush** [comment]
`jspm install npm:jade@1.8.0` breaks
`jspm install npm:jade@1.7.0` works

**Comment by guybedford** [comment]
Thanks, I've added a fix, but it may be a little while before the next release.

**Comment by Rush** [comment]
Thanks for the fix. I can confirm that with the master it does install properly. (Un)fortunately I moved to compiling jade on the server cause our build was failing miserably due to this problem with me being desperate to find a solution. :) I have no reason why it started failing though.

That being said, if it is a major problem like packages not being installed, doesn't it deserve a new release?

**Comment by guybedford** [comment]
Released in 0.16.19.
